homeassistant/components/sensor/prezzibenzina.py

```diff
@@ -0,0 +1,116 @@
+"""
+Support for the PrezziBenzina.it service.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.prezzibenzina/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import (
+    ATTR_ATTRIBUTION, ATTR_TIME, CONF_NAME)
+from homeassistant.helpers.config_validation import PLATFORM_SCHEMA
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['prezzibenzina-py==1.1.4']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_SERVICE = 'service'
+ATTR_FUEL = 'fuel'
+
+CONF_TYPES = 'fuel_types'
+
+ATTRIBUTION = 'Data provided by PrezziBenzina.it'
+
+CONF_STATION = 'station'
+
+SCAN_INTERVAL = timedelta(minutes=120)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_STATION): cv.string,
+    vol.Optional(CONF_NAME, None): cv.string,
+    vol.Optional(CONF_TYPES, None): cv.ensure_list
```

Comment:
How about use `vol.In(['Benzina', 'GPL', ...])` to limit the possibilities?

Reply:
I don't think that's completely doable. PrezziBenzina is mostly crowdsourced so I don't know if the data structure is always the same...

Reply:
Without counting vendor-specific fuels. 
Example: Blue Super, Diesel Speciale, ...

Reply:
I've checked with PrezziBenzina and they currently support: 
`['Benzina', 'Benzina speciale', 'Diesel', 'Diesel speciale', 'GPL', 'Metano']` so vendor-specific fuels get classified as "speciale". 
Adding this to the code.

Reply:
Marking this conversation as solved